src/tests/payment/test_views.py

```diff
@@ -0,0 +1,151 @@
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+        assert response.data == {"detail": "No StripeProduct matches the given query."}
+
+    @patch("payment.views.stripe.checkout.Session.create")
+    def test_create_subscription_stripe_session_creation_failed(
+        self, mock_create_session, client, user, essentiel_product
+    ):
+        client.force_login(user)
+        data = {"plan": "monthly"}
+        mock_create_session.side_effect = Exception("Error during session creation")
+        response = client.post(self.create_url, data, format="json")
+        assert response.status_code == status.HTTP_500_INTERNAL_SERVER_ERROR
+        assert {"error": "Error during session creation"} == response.data
+
+    @patch("payment.views.stripe.checkout.Session.create")
+    def test_retrieve_subscription_success(
+        self, mock_create_session, client, user, essentiel_product
+    ):
+        client.force_login(user)
+        subscription = Subscription.objects.create(
+            user=user, stripe_subscription_id="session_id_example"
+        )
+        response = client.get(self.retrieve_url(subscription.pk), format="json")
+        assert response.status_code == status.HTTP_200_OK
+        assert response.data["user"] == user.id
+        assert response.data["stripe_subscription_id"] == "session_id_example"
+
+    def test_retrieve_subscription_not_found(self, client, user):
+        client.force_login(user)
+        response = client.get(self.retrieve_url(9999), format="json")
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    def test_subscription_success_view(self, client):
+        response = client.get("/api/v1/payment/subscriptions/success/")
+
+        assert response.status_code == status.HTTP_200_OK
+        assert response.data["message"] == "Subscription successful"
+
+    def test_subscription_cancel_view(self, client):
+        response = client.get("/api/v1/payment/subscriptions/cancel/")
+
+        assert response.status_code == status.HTTP_200_OK
+        assert response.data["message"] == "Subscription cancelled"
+
+    def test_subscription_user_cancel_view(self, client, user):
+        client.force_login(user)
+
+        subscription = Subscription.objects.create(
+            user=user,
+            stripe_subscription_id="sub_1FgsVx2R1LZ5sbG0fFqkg9Jz",
+            cancel_at_period_end=False,
+        )
+
+        with mock.patch("stripe.Subscription.modify") as mock_stripe_modify:
+            mock_stripe_modify.return_value = {
+                "id": "sub_1FgsVx2R1LZ5sbG0fFqkg9Jz",
+                "cancel_at_period_end": True,
+            }
+            response = client.post(self.cancel_user_url, data={}, format="json")
+
+        assert response.status_code == status.HTTP_200_OK
+        assert response.data["message"] == "Subscription cancelled successfully"
+
+        subscription.refresh_from_db()
+        assert subscription.cancel_at_period_end is True
```

Comment:
Assert that the subscription status is still active

Reply:
done